tests/suites/upgrade/streams.sh

```diff
@@ -4,16 +4,31 @@ run_simplestream_metadata_last_stable() {
 	jujud_version=$(jujud_version)
 	previous_version=$(last_stable_version "${jujud_version}")
 
-	exec_simplestream_metadata "stable" "${jujud_version}" "${previous_version}"
+	exec_simplestream_metadata "stable" "juju" "${jujud_version}" "${previous_version}"
 }
 
 run_simplestream_metadata_prior_stable() {
 	local jujud_version previous_version
 
 	jujud_version=$(jujud_version)
 	previous_version=$(prior_stable_version "${jujud_version}")
+	major=$(echo "${previous_version}" | cut -d '.' -f 1)
+	minor=$(echo "${previous_version}" | cut -d '.' -f 2)
 
-	exec_simplestream_metadata "prior" "${jujud_version}" "${previous_version}"
+	action=$(snap info juju | grep -q "installed" || echo "install")
+	if [ "${action}" == "" ]; then
+		action="refresh"
+	fi
```

Comment:
```suggestion
	action=$(snap info juju | grep -q "installed" && echo "action" || echo "install")
```
In retrospect, I think you can actually do this, but the if does make it easier to read.